### PR TITLE
MO-7328 Moved the iOS deep link self service file into both Android a…

### PR DESCRIPTION
**Pull request by ndurant-sp**
…nd iOS using a CDN

### CODE REVIEW

File: articles/tech-resources/deep-links-self-serve.md [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+- (BOOL)application:(UIApplication *)application continueUserActivity:(NSUserActivity *)userActivity
+ restorationHandler:(void (^)(NSArray *))restorationHandler {
+
+    // Ignore non-universal link activities
+    if (![userActivity.activityType isEqualToString:NSUserActivityTypeBrowsingWeb] || userActivity.webpageURL == nil) {
+        return NO;
+    }
+
+    // Create a task to call the SparkPost click tracker, expecting a 3xx redirect to your original URL.
+    // On completion, the handler will receive an NSURLResponse which knows the original target URL.
```
Comment (ndurant-sp) [inline articles/tech-resources/deep-links-self-serve.md]: With a CDN you can just do a GET on the original Tracking link, so I removed that substitution but hopefully the objC is still valid.

Comment (ndurant-sp) [reply articles/tech-resources/deep-links-self-serve.md]: More Info:  
With a CDN, it is no longer required to redirect the link to spgo.io as the tracking domain will successfully forward the click.  Direction for this ticket was to focus on the use of a CDN versus a method of removing CNAME method.

**Comment by ndurant-sp** [comment]
MO-7328
- Update the previous ios-self-serve Documentation to incorporate Android App Links
- Moved the logic away from doing a CNAME removal to using a Content Deliver Network (CDN) as this is the more desired method to self serve, and is viable for both SPC and SPE
- Example Android Activity and AsyncTask.

File: articles/tech-resources/deep-links-self-serve.md [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+    }];
+
+    // Kick off the task
+    [clickTask resume];
+
+    return YES;
+}
+```
+
+With a little configuration, a single additional handler in your app, and no additional hosting requirements, you can enable and track iOS universal links through your SparkPost account just as if they were any other link.
```
Comment (ewandennis) [inline articles/tech-resources/deep-links-self-serve.md]: This sentence used to be the last in the article. It was a little weak at the time and it's out of place now. Shall we just remove it?

File: articles/tech-resources/deep-links-self-serve.md [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+
+#### Forwarding Clicks From Android To SparkPost
+
+When an Android email client recognizes that a link has been clicked that is registered via your `AndoidManifest.xml` file.  For examples on configuring an Androdi App Link, see the documenation linked above.  Once the link is clicked, an `intent` is sent to the registered `Activity` to handle the click.  Once inside the activity, the tracking link can be manually "clicked" by performing a HTTP GET request on the link passed in with the `intent`.
```
Comment (ewandennis) [inline articles/tech-resources/deep-links-self-serve.md]: How about:

> When an Android email client recognizes that an app link has been clicked based on your apps' `AndroidManifest.xml`, it sends an `intent` which triggers the registered `Activity` in your app. You can then make an HTTP request to the link to trigger a "click" event in Sparkpost and retrieve the original tracked URL from the message.

File: articles/tech-resources/deep-links-self-serve.md [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+
+Here is a sample `Activity` with the corresponding `AsyncTask` that will perform an HTTP GET to the SparkPost click tracking service after an Android App Link has been clicked:
+```java
+    @Override
+    protected void onCreate(@Nullable Bundle savedInstanceState) {
```
Comment (ewandennis) [inline articles/tech-resources/deep-links-self-serve.md]: Should we include the class that `onCreate` belongs to to people know where to include this code? Maybe just enclose the method in an outline class declaration.

File: articles/tech-resources/deep-links-self-serve.md [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+
+The setup steps are as follows:
+
+1. Configure a CDN to host your custom tracking domain.  One example can be found [here](https://www.sparkpost.com/docs/tech-resources/enabling-https-engagement-tracking-on-sparkpost/).
+**Note:** The CDN needs to support file hosting such as AWS CloudFront.
+1. [Configure a custom tracking domain on your SparkPost account.](https://app.sparkpost.com/account/tracking-domains)
+1. [Use a "custom link sub-path" in your links.](https://developers.sparkpost.com/api/substitutions-reference.html#header-custom-link-sub-paths)
+1. Publish the `apple-app-site-associate` or `assetlinks.json` file for your app using both the tracking domain and link sub-path.  These should be hosted on your CDN under  **\<your-tracking-domain>\\.well-known\\**. Your apple-app-site-association file might include the following:
+    ```javascript
```
Comment (ewandennis) [inline articles/tech-resources/deep-links-self-serve.md]: I wonder if we include example deep link specs for Android.  We could just indent the 2 example code blocks by 4 spaces to have them appear under the numbered list item.

File: articles/tech-resources/deep-links-self-serve.md [hunk 6]
```diff
@@ -0,0 +1,206 @@
...
+The following developer documentation explains the deep linking mechanisims in detail.
+- [Apple's Universal Links](https://developer.apple.com/library/content/documentation/General/Conceptual/AppSearch/UniversalLinks.html) 
+
+- [Android App Links](https://developer.android.com/training/app-links/index.html)
+
+
+
+## Prerequisites
+
+- a moble app with deep linking support
```
Comment (ewandennis) [inline articles/tech-resources/deep-links-self-serve.md]: Typo: mobile

File: articles/tech-resources/deep-links-self-serve.md [hunk 7]
```diff
@@ -0,0 +1,206 @@
...
+
+This will cause the tracked URL path to start with `/f/open-in-app/` which simplifies writing your `apple-app-site-association` file.
+
+### Scenario 1: Universal Links Without Click Tracking
+
+Pros:
+- Very simple
+ 
+Cons:
+- No click tracking on deep linking
```
Comment (ewandennis) [inline articles/tech-resources/deep-links-self-serve.md]: `...on deep links` ?

File: articles/tech-resources/deep-links-self-serve.md [hunk 8]
```diff
@@ -0,0 +1,206 @@
...
+
+### Scenario 2: Deep Links With Click Tracking
+
+Pros:
+- Minimal development effort
+- Provides click tracking on deep links
+
+Cons:
+- Requires mobile app support
+- Configuration of a CDN
```
Comment (ewandennis) [inline articles/tech-resources/deep-links-self-serve.md]: `Requires a CDN` ?

File: articles/tech-resources/deep-links-self-serve.md [hunk 9]
```diff
@@ -0,0 +1,206 @@
...
+
+The setup steps are as follows:
+
+1. Configure a CDN to host your custom tracking domain.  One example can be found [here](https://www.sparkpost.com/docs/tech-resources/enabling-https-engagement-tracking-on-sparkpost/).
+**Note:** The CDN needs to support file hosting such as AWS CloudFront.
+1. [Configure a custom tracking domain on your SparkPost account.](https://app.sparkpost.com/account/tracking-domains)
+1. [Use a "custom link sub-path" in your links.](https://developers.sparkpost.com/api/substitutions-reference.html#header-custom-link-sub-paths)
+1. Publish the deep linking spec file for your app using both the tracking domain and link sub-path.  These should be hosted on your CDN under  **\<your-tracking-domain>\\.well-known\\**.
```
Comment (ewandennis) [inline articles/tech-resources/deep-links-self-serve.md]: `This should be hosted...` ?

**Pull request merged**